app/scripts/models/auth_brokers/redirect.js

```diff
@@ -17,14 +17,22 @@ define([
       return p()
         .then(function () {
           var redirectTo = result.redirect;
+          var separator;
 
           if (result.error) {
             // really, we should be parsing the URL and adding the error
             // parameter. That requires more code than this.
-            var separator = redirectTo.indexOf('?') === -1 ? '?' : '&';
+            separator = redirectTo.indexOf('?') === -1 ? '?' : '&';
```

Comment:
Maybe it's time to parse the URI, then make use of [lib.js->objToSearchString](https://github.com/mozilla/fxa-content-server/blob/4898dc006863fe26746b7562cc5dad78c36ca2f5/app/scripts/lib/url.js#L49), this is starting to look a bit out of control. Not sure how much more effort that would be.